minmea.c

```diff
@@ -612,25 +612,33 @@ bool minmea_parse_zda(struct minmea_sentence_zda *frame, const char *sentence)
   return true;
 }
 
-int minmea_gettime(struct timespec *ts, const struct minmea_date *date, const struct minmea_time *time_)
+int minmea_getdate(struct tm *tm, const struct minmea_date *date, const struct minmea_time *time_)
 {
     if (date->year == -1 || time_->hours == -1)
         return -1;
 
-    struct tm tm;
-    memset(&tm, 0, sizeof(tm));
+    memset(tm, 0, sizeof(*tm));
```

Comment:
can we use the type here and not dereference the pointer? I'd be worried sizeof() would be taking a sizeof() a pointer which is 4 and not doing the right thing here.

Reply:
The pointer dereference here is the correct and safe thing to do - it will follow in case the variable type changes.